demos/pedestrian_tracker_demo/main.cpp

```diff
@@ -158,35 +152,39 @@ int main_work(int argc, char **argv) {
         CreatePedestrianTracker(reid_model, reid_weights, ie, reid_mode,
                                 should_keep_tracking_info);
 
-
-    // Opening video.
-    std::unique_ptr<ImageReader> video =
-        ImageReader::CreateImageReaderForPath(video_path);
-
-    PT_CHECK(video->IsOpened()) << "Failed to open video: " << video_path;
-    double video_fps = video->GetFrameRate();
-
-    if (first_frame > 0)
-        video->SetFrameIndex(first_frame);
+    cv::VideoCapture cap;
+    try {
+        int intInput = std::stoi(FLAGS_i);
+        if (!cap.open(intInput)) {
+            throw std::runtime_error("Can't open " + std::to_string(intInput));
+        }
+    } catch (const std::invalid_argument&) {
+        if (!cap.open(FLAGS_i)) {
+            throw std::runtime_error("Can't open " + FLAGS_i);
+        }
+    } catch (const std::out_of_range&) {
+        if (!cap.open(FLAGS_i)) {
+            throw std::runtime_error("Can't open " + FLAGS_i);
+        }
+    }
+    double video_fps = cap.get(cv::CAP_PROP_FPS);
+    if (0.0 == video_fps) {
+        // the default frame rate for DukeMTMC dataset
+        video_fps = 60.0;
+    }
+    if (0 != FLAGS_first && !cap.set(cv::CAP_PROP_POS_FRAMES, FLAGS_first)) {
+        throw std::runtime_error("Can't set the frame to begin with");
+    }
 
     std::cout << "To close the application, press 'CTRL+C' here";
     if (!FLAGS_no_show) {
         std::cout << " or switch to the output window and press ESC key";
     }
     std::cout << std::endl;
 
-    for (;;) {
-        auto pair = video->Read();
-        cv::Mat frame = pair.first;
-        int frame_idx = pair.second;
-
-        if (frame.empty()) break;
-
-        PT_CHECK(frame_idx >= first_frame);
-
-        if ( (last_frame >= 0) && (frame_idx > last_frame) ) {
-            std::cout << "Frame " << frame_idx << " is greater than last_frame = "
-                << last_frame << " -- break";
+    for (uint32_t frame_idx = FLAGS_first; 0 == FLAGS_last || frame_idx <= FLAGS_last; ++frame_idx) {
```

Comment:
Considering that 0 is a valid frame index, I think it would be better if only negative `FLAGS_last` values meant "process all frames".

Reply:
I think we should keep `FLAGS_first` and `FLAGS_last` of the same type just for consistency and clarity. So I can 
1) revert both flags back to `int32`

or

2. reformulate `FLAGS_last` as `The frame index of video sequence to stop at.` And `FLAGS_first >= FLAGS_last` will mean process nothing (instead of `FLAGS_first > FLAGS_last`), and `FLAGS_last=0` makes the demo stop immediately. 

What do you think?

Reply:
I think option 1 makes the most sense.

Reply:
updated